lib/instance.js

```diff
@@ -342,6 +342,10 @@ Instance.prototype.set = function(key, value, options) { // testhint options:non
           if (this.Model._hasDateAttributes && this.Model._isDateAttribute(key) && value !== null && value !== undefined && !(value instanceof Date) && !value._isSequelizeMethod) {
             value = new Date(value);
           }
+
+          if (Utils._.isDate(value) && Utils._.isDate(originalValue) && value.getTime() === originalValue.getTime()) {
+            return;
+          }
```

Comment:
This could probably be combined with the previous if.
So we'd have one wrapper if checking `isDateAttribute` and then two subsequent ifs making it a `Date` if it isn't and then comparing.